### ISSUE TITLE
Cannot compile Hello World

### ISSUE THREAD

**Issue by Aetet** [maps4print/azul#189]
## Description
Cannot execute `cargo run`, because it fails with following mistake:
```
cargo run --verbose
    Updating git repository `https://github.com/maps4print/azul`
    Updating git repository `https://github.com/maps4print/azul-dependencies`
    Updating crates.io index
    Updating git repository `https://github.com/fschutt/simplecss.git`
error: failed to load source for a dependency on `simplecss`

Caused by:
  Unable to update https://github.com/fschutt/simplecss.git?rev=0bf3b7800f49a4524b889687d7761b163e8b4101

Caused by:
  revspec '0bf3b7800f49a4524b889687d7761b163e8b4101' not found; class=Reference (4); code=NotFound (-3)

```

## Version / OS
* azul version:
```
[package]
name = "my_first_azul_app"
version = "0.1.0"
authors = ["Aetet <aetetic@gmail.com>"]

[dependencies]
azul = { git = "https://github.com/maps4print/azul" }

```

* Operating system:
Linux 4.15.0-36-generic
Ubuntu 18.04.1 LTS 

* Windowing system - X11

## Steps to Reproduce
Create hello world
```rust
extern crate azul;

use azul::prelude::*;

struct MyDataModel { }

impl Layout for MyDataModel {
    fn layout(&self, _: LayoutInfo<Self>) -> Dom<Self> {
        Dom::div()
    }
}

fn main() {
    let mut app = App::new(MyDataModel { }, AppConfig::default()).unwrap();
    let window = app.create_window(WindowCreateOptions::default(), css::native()).unwrap();
    app.run(window).unwrap();
}

```

**Comment by fschutt** [comment]
yes, it's because I force-pushed on fschutt/simplecss, which destroyed the git commit.

The maintainer of simplecss didn't want to merge my patches, so I had to upload it as [azul-simplecss](https://crates.io/crates/azul-simplecss). It's already fixed in the `/unvendor_dependencies` branch, but that branch isn't ready yet (need to update webrender because of panics on Windows).

Should be fixed when the `/unvendor_dependencies` branch gets merged.

**Comment by Phil-dev17** [comment]
Having the same problem on OSX, any solution but to wait on the /unvendored_dependencies?

**Comment by abritinthebay** [comment]
Any ETA on a fix for this? Seems a pretty major break. Would be good to know when that branch is going to get merged.

**Comment by jlgerber** [comment]
@fschutt, it would be really helpful if you started using semantic versioning, tagging your releases. 

That way, we could all happily reference the version prior to breaking the trunk.

**Comment by fschutt** [comment]
@jlgerber Yeah, it's just that there is no release yet (it's all way too unstable). Plus, since force-pushing destroyed that commit, version tagging doesn't help. I did not expect people to rely on azul yet, since most of the APIs are changing every few days.

If you rely on azul, **vendor / copy everything and lock your versions with commit hashes**. Don't rely on github repos to never be deleted or go down (and also don't trust crates.io to do the same, but it's less likely that cargo screws up, generally everything that is on crates.io should build).

I've fixed this issue now temporarily in https://github.com/maps4print/azul/commit/32a3f11892ccbfa8baa6c32ce5c48f75c4fbb076, since the `unvendor_dependencies` branch still needs work (I upgraded glutin and webrender which will fix a lot of bugs, but that upgrading broke everything else in the process, lots of segfaults everywhere with some OpenGL errors). It's a work in progress.

**Comment by abritinthebay** [comment]
Dang, I'd got to that point myself (forked it to play) and unfortunately it *builds* but won't run.

I get those GL segfaults you mentioned.

```
[ERROR][azul::logging] An unexpected panic ocurred, the program has to exit.
Please report this error and attach the log file found in the directory of the executable.

The error ocurred in: /projects/.cargo/git/checkouts/azul-dependencies-70bb1f94316762f9/bf2933b/tomaka.glutin#23b3b10/src/platform/macos/mod.rs at line 61 in thread main

Error information:
not yet implemented

Backtrace:

azul::logging::set_up_panic_hooks::panic_fn @ logging.rs:86
core::ops::function::Fn::call @ function.rs:69
std::panicking::rust_panic_with_hook @ panicking.rs:478
std::panicking::begin_panic @ panicking.rs:408
glutin::platform::platform::Context::new @ mod.rs:61
glutin::combined::CombinedContext::new @ combined.rs:58
azul::window::create_gl_window @ window.rs:890
azul::window::Window<T>::new @ window.rs:483
azul::app::App<T>::create_window @ app.rs:211
megablast::main @ main.rs:26
std::rt::lang_start::{{closure}} @ rt.rs:64
std::panicking::try::do_call @ panicking.rs:293
 @ lib.rs:85
std::rt::lang_start_internal @ rt.rs:48
std::rt::lang_start @ rt.rs:64
```

**Comment by fschutt** [comment]
@abritinthebay Those are already fixed on the unvendor_dependencies branch. On master, context sharing is not implemented on macOS, see #131.
